server/src/main/java/org/opensearch/snapshots/SnapshotUtils.java

```diff
@@ -89,7 +103,7 @@ public static List<String> filterIndices(List<String> availableIndices, String[]
                         result = new HashSet<>();
                     }
                 } else if (indexOrPattern.charAt(0) == '-') {
-                    // if its the first, fill it with all the indices...
+                    // if its the first index pattern, fill it with all the indices...
```

Comment:
I'd change this comment to something like:

> If the first index pattern is an exclusion, then all patterns are exclusions due to the reordering logic above. In this case, the request is interpreted as "include all indexes except those matching the exclusions" so we add all indices here and then remove the ones that match the exclusion patterns.

Reply:
Will do!